pkg/helmreconciler/rendering.go

```diff
@@ -220,13 +220,14 @@ func (h *HelmReconciler) ProcessObject(chartName string, obj *unstructured.Unstr
 	gvk := mutatedObj.GetObjectKind().GroupVersionKind()
 	receiver.SetGroupVersionKind(gvk)
 	objectKey, _ := client.ObjectKeyFromObject(mutatedObj)
-	namespacedResourceMap, nonNamespacedResourceMap := h.customizer.PruningDetails().GetResourceTypes()
-	if val, ok := namespacedResourceMap[gvk]; !val && ok {
+	namespacedResourceMap, nonNamespacedResourceMap, pruningDetailsMU := h.customizer.PruningDetails().GetResourceTypes()
+	pruningDetailsMU.Lock()
```

Comment:
We only need to update the map when h.needUpdateAndPrune == true.

Reply:
Thanks, done.